### PR TITLE
ROX-23946: Factor out email template from vulnerability reporting

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

Prerequisite to factor out delivery destinations from Vulnerability Reports to reuse in Compliance Reports.

### Changes

1. Create EmailTemplate.utils.ts
    * Move maximum lengths from useReportFormValues.tsx file and leave what is specific to vulnerability reports.
    * Move reusable validations from emailTemplateFormUtils.ts file and leave what is specific to vulnerability reports.
    * Rename variables for consistency with `NotifierConfiguration` type.
2. Create EmailTemplateForm.tsx
    * Rename variables for consistency with `NotifierConfiguration` type.
3. Move and rename EmailTemplateModal.tsx
    * Replace `reportParameters` with optional `renderTemplatePreview` prop.
    * Rename variables for consistency with `NotifierConfiguration` type.
    * Factor out form.
    * Conditionally render tabs.
4. Edit DeliveryDestinationsForm.tsx
    * Import from `Components/EmailTemplate` folder.
    * Factor out `renderTemplatePreview` which is specific to vulnerability reports.
5. Edit emailTemplateFormUtils.ts
    * Move reusable validations.
6. Edit useReportFormValues.tsx
    * Move reusable lengths and validations.

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

## Testing Performed

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js 0 = 4633558 - 4633558
        total 353 = 11601343 - 11600990
    * `ls -al build/static/js/*.js | wc`
        files 0 = 173 - 173
3. `yarn start` in ui

### Manual testing

Prerequisite: create at least one collection.

Create e-mail notifiers on the fly.

1. Create 3 reports with various combinations of defaut or custom destinations or template.
    * Verify payload and response of POST /v1/notifiers request.
    * Verify review details.
    * Verify payload and response of POST /v2/reports/configurations request.
2. Edit the first report and ditto items 2 and 3 above.

EmailTemplateModal **Edit** tab
EmailTemplateModal_1_Edit

EmailTemplateModal **Preview** tab and **CVEs found** toggle
EmailTemplateModal_1_Preview_CVEs_found

EmailTemplateModal **Preview** tab and **CVEs not found** toggle
EmailTemplateModal_1_Preview_CVEs_not_found

EmailTemplateModal with EmailTemplateForm only (that is, without `renderTemplatePreview` prop, which compliance reports probably not have)
EmailTemplateModal_with_EmailTemplateForm_only

DeliveryDestinationsDetails
DeliveryDestinationsDetails

EmailTemplatePreview
EmailTemplatePreview

### Integration testing

Uh, oh! No tests. Yet.

### CODE REVIEW

File: ui/apps/platform/src/Components/EmailTemplate/EmailTemplateForm.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    maxCustomBodyLength,
+    maxCustomSubjectLength,
+} from './EmailTemplate.utils';
+
+function getFieldValidated(
+    errors: FormikErrors<EmailTemplateFormData>,
+    touched: FormikTouched<EmailTemplateFormData>,
+    fieldId: string
+) {
+    const isFieldInvalid = !!(get(errors, fieldId) && get(touched, fieldId));
```
[the remark this answers is not in the document]
Comment (pedrottimark) [reply ui/apps/platform/src/Components/EmailTemplate/EmailTemplateForm.tsx:30]: Good eyes! This is a great question which I have asked also.
* Not benefit for `customBody` and `customSubject` at first level.
* But there is a benefit for second level or arrays like here: https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/DeliveryDestinationsForm.tsx#L169

I will refactor the helper function to use the idiom that you suggest.

File: ui/apps/platform/src/Components/EmailTemplate/EmailTemplateForm.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                />
+                <FormHelperText>
+                    <HelperText>
+                        <HelperTextItem variant={isSubjectValidated}>
+                            {errors.customSubject}
+                        </HelperTextItem>
+                    </HelperText>
+                </FormHelperText>
+                <Flex>
+                    <FlexItem flex={{ default: 'flex_1' }}>
```
Comment (bradr5) [inline ui/apps/platform/src/Components/EmailTemplate/EmailTemplateForm.tsx:70]: Not a change request just wanted to point out that this patternfly prop always confuses me. I feel like using either `flex-grow` here of `align-self: end` on the button feel more intuitive to me. Or even `justify-content: space-between` but that could cause more issues if another element is added.

**Review by bradr5 — APPROVED** [review]

**Pull request merged**